xml/System/UriBuilder.xml

```diff
@@ -975,22 +981,27 @@
-  
- The query information is escaped according to RFC 2396.  
-  
- **Note** In version 1.0 and 1.1 of the .NET Framework, setting the <xref:System.UriBuilder.Fragment%2A> property to any value, including `null`, cleared the <xref:System.UriBuilder.Query%2A> property, and setting the <xref:System.UriBuilder.Fragment%2A> property to `null` or to <xref:System.String.Empty?displayProperty=nameWithType> cleared the property. In version 2.0, however, the <xref:System.UriBuilder.Fragment%2A> and <xref:System.UriBuilder.Query%2A> properties are independent.  
-  
- **Note** Do not append a string directly to this property. If the length of <xref:System.UriBuilder.Query%2A> is greater than 1, retrieve the property value as a string, remove the leading question mark, append the new query string, and set the property with the combined string.  
-  
-   
-  
-## Examples  
- The following example sets the <xref:System.UriBuilder.Query%2A> property.  
-  
- :::code language="cpp" source="~/samples/snippets/cpp/VS_Snippets_Remoting/UriBuilderSample/cpp/main.cpp" id="Snippet1":::
- :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_Remoting/UriBuilderSample/cs/main.cs" id="Snippet1":::  
+## Remarks
+
+The <xref:System.Uri.Query> property contains any query information included in the URI. Query information is separated from the path information by a question mark (?) and continues to the end of the URI. The query information that's returned includes the leading question mark. When setting the <xref:System.UriBuilder.Query> property:
+
+- In .NET Framework, a question mark is always prepended to the string.
+- In .NET Core and later versions, a question mark is prepended to the string if it's not already present.
+
+The query information is escaped according to RFC 2396.  
+
+> [!NOTE]
+> In version 1.0 and 1.1 of .NET Framework, setting the <xref:System.UriBuilder.Fragment> property to any value, including `null`, cleared the <xref:System.UriBuilder.Query> property, and setting the <xref:System.UriBuilder.Fragment> property to `null` or to <xref:System.String.Empty?displayProperty=nameWithType> cleared the property. In .NET Framework 2.0 and later versions, however, the <xref:System.UriBuilder.Fragment> and <xref:System.UriBuilder.Query> properties are independent.  
+
+> [!NOTE]
+> When appending a string to this property, there's a behavior difference between .NET Framework and later versions of .NET to be aware of. In .NET (including .NET Core), you can simply retrieve the property, append the new string, and then set the property. However, in .NET Framework, if the length of <xref:System.UriBuilder.Query> is greater than 1, you must remove the leading question mark before appending the new query string and setting the property.
```

Comment:
```suggestion
> To append a value in .NET 5 (and .NET Core) and later versions, you can retrieve the property value, append the new string, and then set the property. In .NET Framework, if the length of <xref:System.UriBuilder.Query> is greater than 1, you must remove the leading question mark before appending the new query string and setting the property.
```

Reply:
```suggestion
> To append a value to existing query information in .NET Framework, you must remove the leading question mark before setting the property with the new value. This is because .NET Framework always prepends the question mark when setting the property. .NET 5 (and .NET Core) and later versions are tolerant to a leading question mark, and will only prepend one if necessary.
```